modules/python/test/test_unicode_support.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding:utf-8 -*-
+from __future__ import print_function
+
+import numpy as np
+import cv2 as cv
+import os
+import tempfile
+import sys
+
+from tests_common import NewOpenCVTests
+
+str_unicode = u"测试tést"
+
+
+def rmdir(directory):
+    for root, dirs, files in os.walk(directory, topdown=False):
+        for name in files:
+            os.remove(os.path.join(root, name))
+        for name in dirs:
+            os.rmdir(os.path.join(root, name))
+    os.rmdir(directory)
+
+
+class unicode_support_test(NewOpenCVTests):
+    tmp_path = None
+
+    def setUp(self):
+        if os.name != 'nt' and sys.getfilesystemencoding() != "utf-8":
+            self.skipTest("environment variable `LANG` is not utf8")
+            return
+        self.tmp_path = tempfile.mkdtemp(str_unicode)
+
+    def tearDown(self):
+        rmdir(self.tmp_path)
+
+    def test_fs_get_cache_dir(self):
+        if os.name != 'nt':
+            self.skipTest("only windows")
+            return
+        tmp_bak = os.environ["TMP"]
+        os.environ["TMP"] = self.tmp_path
+
+        path = cv.utils.fs.getCacheDirectoryForDownloads()
+        self.assertTrue(os.path.exists(path))
+        self.assertTrue(os.path.isdir(path))
+        self.assertIn(self.tmp_path, path)
+
+        os.environ["TMP"] = tmp_bak
+
+    def test_im_rw(self):
+        data_write = np.random.randint(0, 255, (4, 4, 3))
+        file_path = os.path.join(self.tmp_path, str_unicode + ".png")
+        cv.imwrite(file_path, data_write)
+        self.assertTrue(os.path.exists(file_path))
+        data_read = cv.imread(file_path, cv.IMREAD_UNCHANGED)
+        self.assertTrue(np.array_equal(data_write, data_read))
+
+    def test_video_write(self):
```

Comment:
I think this check is redundant.

Conversions between OpenCV -> Python -> OpenCV are already checked in `getCacheDirectory` and `imread/imwrite` tests.

Reply:
Yes! I just want to make the unicode tests more robust to prevent bugs elsewhere or regressions that might be introduced in the future.